src/core/net/nd6.cpp

```diff
@@ -179,6 +179,8 @@ void RaFlagsExtOption::Init(void)
     Clear();
     SetType(kTypeRaFlagsExtension);
     SetSize(sizeof(RaFlagsExtOption));
+
+    OT_UNUSED_VARIABLE(mFlags);
```

Comment:
Yes. We can get warnings on this, since it is `private` and never directly access by any method (though it is indirectly set using `Clear()`). 

We have similar situation in a bunch of other TLV/Option definitions. For example:
```c++
void PrefixInfoOption::Init(void)
{
    Clear();
    SetType(kTypePrefixInfo);
    SetSize(sizeof(PrefixInfoOption));

    OT_UNUSED_VARIABLE(mReserved2);
}
```